client_linux.go

```diff
@@ -139,3 +163,60 @@ func parsePorts(msgs []genetlink.Message) ([]*Port, error) {
 
 	return ps, nil
 }
+
+// parseDpipeTables parses DPIPE tables from a slice of generic netlink messages.
+func parseDpipeTables(msgs []genetlink.Message) ([]*DpipeTable, error) {
+	var bus, dev string
+	if len(msgs) == 0 {
+		// No devlink response found.
+		return nil, nil
+	}
+
+	ts := make([]*DpipeTable, 0, len(msgs))
+	for _, m := range msgs {
+		ad, err := netlink.NewAttributeDecoder(m.Data)
+		if err != nil {
+			return nil, err
+		}
+
+		for ad.Next() {
+			switch ad.Type() {
+			case unix.DEVLINK_ATTR_BUS_NAME:
+				bus = ad.String()
+			case unix.DEVLINK_ATTR_DEV_NAME:
+				dev = ad.String()
+			case unix.DEVLINK_ATTR_DPIPE_TABLES:
+				tablesData := ad.Bytes()
```

Comment:
This is a good place to use `ad.Nested`. And since the code is already pretty heavily indented, I think most of this switch case should be factored out into its own function. You can see some of the patterns I've worked out for doing this sort of thing here: https://github.com/WireGuard/wgctrl-go/blob/master/internal/wglinux/parse_linux.go.

Reply:
Thank you! I used you example to improve the code.